Build failure with #4776

- distro: arch
- package: `waybar-git`
- compiler: `g++ (GCC) 15.2.1 20260103`
- compiler: `clang version 21.1.6`

With #4776 merged, I'm getting build failure when building the aur package.

<details><summary>terminal output</summary>

```
(git)-[master]-% meson setup build --reconfigure -Dcava=disabled -Dgps=disabled -Dtests=disabled
(git)-[master]- [130]% ninja -C build
ninja: Entering directory `build'
[71/194] Compiling C++ object waybar.p/src_bar.cpp.o
FAILED: [code=1] waybar.p/src_bar.cpp.o 
/usr/bin/sccache c++ -Iwaybar.p -I. -I.. -I../include -Iresources/icons -Iprotocol -I/usr/include/sigc++-2.0 -I/usr/lib/sigc++-2.0/include -I/usr/include/gtkmm-3.0 -I/usr/lib/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/lib/atkmm-1.6/include -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/giomm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/cloudproviders -I/usr/include/cairo -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/atk-1.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/pangomm-1.4 -I/usr/lib/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/glibmm-2.4/include -I/usr/include/cairomm-1.0 -I/usr/lib/cairomm-1.0/include -I/usr/include/pango-1.0 -I/usr/include/pixman-1 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glycin-2 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/sysprof-6 -I/usr/include/libdbusmenu-gtk3-0.4 -I/usr/include/libdbusmenu-glib-0.4 -I/usr/include/libnl3 -I/usr/include/libupower-glib -I/usr/include/polkit-1 -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -I/usr/include/playerctl -I/usr/include/wireplumber-0.5 -I/usr/include/libevdev-1.0 -I/usr/include/gtk-layer-shell -I/usr/include/libxml2 -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++20 '-DVERSION="0.14.0-152-gddab2971 (branch '"'"'master'"'"')"' -DHAVE_LANGINFO_1STDAY '-DSYSCONFDIR="/etc"' -DHAVE_CPU_LINUX -DHAVE_MEMORY_LINUX -DHAVE_SYSTEMD_MONITOR -DHAVE_SWAY -DHAVE_WLR_TASKBAR -DHAVE_EXT_WORKSPACES -DHAVE_RIVER -DHAVE_DWL -DHAVE_HYPRLAND -DHAVE_NIRI -DHAVE_WAYFIRE -DHAVE_LOGIN_PROXY -DHAVE_LIBNL -DHAVE_GAMEMODE -DHAVE_LOGIND_INHIBITOR -DHAVE_UPOWER -DHAVE_PIPEWIRE -DHAVE_MPRIS -DHAVE_LIBPULSE -DHAVE_LIBJACK -DHAVE_LIBWIREPLUMBER -DHAVE_DBUSMENU -DHAVE_LIBUDEV -DHAVE_LIBEVDEV -DHAVE_LIBINPUT -DHAVE_LIBMPDCLIENT -DHAVE_LIBSNDIO -DWANT_RFKILL -DHAVE_CHRONO_TIMEZONES -fPIE -pthread -D_REENTRANT -DSPDLOG_SHARED_LIB -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -MD -MQ waybar.p/src_bar.cpp.o -MF waybar.p/src_bar.cpp.o.d -o waybar.p/src_bar.cpp.o -c ../src/bar.cpp
../src/bar.cpp: In constructor ‘waybar::Bar::Bar(waybar::waybar_output*, const Json::Value&)’:
../src/bar.cpp:232:65: error: cannot convert ‘Json::String’ {aka ‘std::__cxx11::basic_string<char>’} to ‘const char*’
  232 |   gtk_layer_set_namespace(gtk_window, config["name"].isString() ? config["name"].asString() : "waybar");
      |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                 |
      |                                                                 Json::String {aka std::__cxx11::basic_string<char>}
In file included from ../src/bar.cpp:3:
/usr/include/gtk-layer-shell/gtk-layer-shell.h:184:62: note: initializing argument 2 of ‘void gtk_layer_set_namespace(GtkWindow*, const char*)’
  184 | void gtk_layer_set_namespace (GtkWindow *window, char const* name_space);
      |                                                  ~~~~~~~~~~~~^~~~~~~~~~
[73/194] Compiling C++ object waybar.p/src_factory.cpp.o
In file included from /usr/include/wireplumber-0.5/wp/wp.h:25,
                 from ../include/modules/wireplumber.hpp:4,
                 from ../src/factory.cpp:109:
/usr/include/wireplumber-0.5/wp/log.h: In function ‘gboolean wp_log_topic_is_enabled(WpLogTopic*, GLogLevelFlags)’:
/usr/include/wireplumber-0.5/wp/log.h:112:24: warning: bitwise operation between different enumeration types ‘WpLogTopicFlags’ and ‘GLogLevelFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  112 |   return (topic->flags & log_level & WP_LOG_TOPIC_LEVEL_MASK) != 0;
      |           ~~~~~~~~~~~~~^~~~~~~~~~~
[88/194] Compiling C++ object waybar.p/src_modules_battery.cpp.o
ninja: build stopped: subcommand failed.
```

</details>

<details><summary>clang++</summary>

```
(git)-[master]-% ninja -C build                                                                 
ninja: Entering directory `build'
[75/193] Compiling C++ object waybar.p/src_bar.cpp.o
FAILED: [code=1] waybar.p/src_bar.cpp.o 
clang++ -Iwaybar.p -I. -I.. -I../include -Iresources/icons -Iprotocol -I/usr/include/sigc++-2.0 -I/usr/lib/sigc++-2.0/include -I/usr/include/gtkmm-3.0 -I/usr/lib/gtkmm-3.0/include -I/usr/include/atkmm-1.6 -I/usr/lib/atkmm-1.6/include -I/usr/include/gtk-3.0/unix-print -I/usr/include/gdkmm-3.0 -I/usr/lib/gdkmm-3.0/include -I/usr/include/giomm-2.4 -I/usr/lib/giomm-2.4/include -I/usr/include/gtk-3.0 -I/usr/include/cloudproviders -I/usr/include/cairo -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/atk-1.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/pangomm-1.4 -I/usr/lib/pangomm-1.4/include -I/usr/include/glibmm-2.4 -I/usr/lib/glibmm-2.4/include -I/usr/include/cairomm-1.0 -I/usr/lib/cairomm-1.0/include -I/usr/include/pango-1.0 -I/usr/include/pixman-1 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glycin-2 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/sysprof-6 -I/usr/include/libdbusmenu-gtk3-0.4 -I/usr/include/libdbusmenu-glib-0.4 -I/usr/include/libnl3 -I/usr/include/libupower-glib -I/usr/include/polkit-1 -I/usr/include/pipewire-0.3 -I/usr/include/spa-0.2 -I/usr/include/playerctl -I/usr/include/wireplumber-0.5 -I/usr/include/libevdev-1.0 -I/usr/include/gtk-layer-shell -I/usr/include/libxml2 -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++20 -O3 '-DVERSION="0.14.0-152-gddab2971 (branch '"'"'master'"'"')"' -DHAVE_LANGINFO_1STDAY '-DSYSCONFDIR="/usr/local/etc"' -DHAVE_CPU_LINUX -DHAVE_MEMORY_LINUX -DHAVE_SYSTEMD_MONITOR -DHAVE_SWAY -DHAVE_WLR_TASKBAR -DHAVE_EXT_WORKSPACES -DHAVE_RIVER -DHAVE_DWL -DHAVE_HYPRLAND -DHAVE_NIRI -DHAVE_WAYFIRE -DHAVE_LOGIN_PROXY -DHAVE_LIBNL -DHAVE_GAMEMODE -DHAVE_LOGIND_INHIBITOR -DHAVE_UPOWER -DHAVE_PIPEWIRE -DHAVE_MPRIS -DHAVE_LIBPULSE -DHAVE_LIBJACK -DHAVE_LIBWIREPLUMBER -DHAVE_DBUSMENU -DHAVE_LIBUDEV -DHAVE_LIBEVDEV -DHAVE_LIBINPUT -DHAVE_LIBMPDCLIENT -DHAVE_LIBSNDIO -DHAVE_CHRONO_TIMEZONES -pthread -D_REENTRANT -DSPDLOG_SHARED_LIB -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -MD -MQ waybar.p/src_bar.cpp.o -MF waybar.p/src_bar.cpp.o.d -o waybar.p/src_bar.cpp.o -c ../src/bar.cpp
../src/bar.cpp:232:3: error: no matching function for call to 'gtk_layer_set_namespace'
  232 |   gtk_layer_set_namespace(gtk_window, config["name"].isString() ? config["name"].asString() : "waybar");
      |   ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-layer-shell/gtk-layer-shell.h:184:6: note: candidate function not viable: no known conversion from 'String' (aka 'basic_string<char, char_traits<char>, std::allocator<char>>') to 'const char *' for 2nd argument
  184 | void gtk_layer_set_namespace (GtkWindow *window, char const* name_space);
      |      ^                                           ~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[76/193] Compiling C++ object waybar.p/src_factory.cpp.o
In file included from ../src/factory.cpp:109:
In file included from ../include/modules/wireplumber.hpp:4:
In file included from /usr/include/wireplumber-0.5/wp/wp.h:25:
/usr/include/wireplumber-0.5/wp/log.h:112:24: warning: bitwise operation between different enumeration types ('WpLogTopicFlags' and 'GLogLevelFlags') is deprecated [-Wdeprecated-enum-enum-conversion]
  112 |   return (topic->flags & log_level & WP_LOG_TOPIC_LEVEL_MASK) != 0;
      |           ~~~~~~~~~~~~ ^ ~~~~~~~~~
1 warning generated.
[92/193] Compiling C++ object waybar.p/src_modules_battery.cpp.o
ninja: build stopped: subcommand failed.
```

</details>

Looks like `gtk_layer_set_namespace` need `char const*` data type.
[Documentation](https://wmww.github.io/gtk-layer-shell/gtk-layer-shell-Gtk-Layer-Shell.html#gtk-layer-set-namespace)

And `Json::Value` [`asString`](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_value.html#a52207c8d4e86160f968a40817cb2529b) function convert data to [Json::String](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/namespace_json.html#a279e78539f3df538c6e3bd36b81d5f3c)(std::basic_string) instead.
[jsoncpp ref](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_value.html#a52207c8d4e86160f968a40817cb2529b)

Switching to [asCString. ref](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_value.html#a16668c8db7ef0a5de040012f0dfd84b0) build

--- a/src/bar.cpp
+++ b/src/bar.cpp
@@ -229,7 +229,7 @@ waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
   gtk_layer_init_for_window(gtk_window);
   gtk_layer_set_keyboard_mode(gtk_window, GTK_LAYER_SHELL_KEYBOARD_MODE_NONE);
   gtk_layer_set_monitor(gtk_window, output->monitor->gobj());
-  gtk_layer_set_namespace(gtk_window, config["name"].isString() ? config["name"].asString() : "waybar");
+  gtk_layer_set_namespace(gtk_window, config["name"].isString() ? config["name"].asCString() : "waybar");
 
   gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_LEFT, margins_.left);
   gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_RIGHT, margins_.right);

